src/routes/TraderForm/Main/PositionDetails/index.tsx

```diff
@@ -16,8 +23,32 @@ import {
 type PositionDetailsProps = {};
 
 export const PositionDetails: React.FunctionComponent<PositionDetailsProps> = () => {
+  const { aMM } = useSwapFormAMM();
+  const notional = useAppSelector(selectNotionalAmount);
+  const variableRateInfo = useAppSelector(selectVariableRateInfo);
+  const infoPostSwap = useAppSelector(selectInfoPostSwap);
   const mode = useAppSelector(selectMode);
 
+  const fixedRate =
+    infoPostSwap.status === 'success' ? formatNumber(infoPostSwap.value.averageFixedRate) : '--';
+  const variableRate =
+    variableRateInfo.status === 'success' ? formatNumber(variableRateInfo.value) : '--';
+
+  const receivingRate = mode === 'fixed' ? fixedRate : variableRate;
+  const payingRate = mode === 'fixed' ? variableRate : fixedRate;
+
+  const compactNotional = compactFormat(stringToBigFloat(notional.value));
+  let notionalBoxToken = (compactNotional.at(-1) || '').match(/[a-z]/i)
```

Comment:
can you explain this logic a bit? compactNotional.at(-1) what does it mean? It is kinda hard to read this. Also using regex during render is not really optimal. Regex is expensive computation. Perhaps we can write a simpler code. What are you trying to do here actually?

Reply:
`compactNotional.at(-1)` is getting the last character to check if it's a letter or not. Let me write an util function that replaces the regex-based isLetter check with an easier implementation.

Reply:
to my mind it kinda feels complex what we are doing to get this 'K', 'M' value and so on

from my analyis problem is 

```
export const compactFormat = (value: number): string => {
  const formatter = Intl.NumberFormat(navigator.language, {
    notation: 'compact',
    maximumFractionDigits: 2,
    minimumFractionDigits: 0,
  });
  return formatter.format(value);
};
```

it doesn't really give us information regarding this letter K
what if we created another function
call it perhaps `compactFormatNumber`

and instead of returning the plain string (formatted) we include the letter as well

```
export const compactFormatNumber = (value: number): string => {
  const formatted = compactFormat(value);
  const lastChar = formatted.at(-1);
  if(isNaN(parseInt(lastChar, 10))) {
    return {
      formatted,
      suffix: lastChar
    }
  }

  return {
    formatted,
    suffix: ''
  }
};
```

now if we try and use this one, we get more information that we can plug into our token components
without having to write complex logic for `isASCIILetter`

we just say const { formatted, suffix } = compatFormatNumber(1111);
// suffix = k, formatted = 1.1
then we pass this to our components

Reply:
Addressed in [3a60d00](https://github.com/Voltz-Protocol/voltz-ui/pull/572/commits/3a60d00a29e7a8c2372ab8cb8aac3d48d7c4c948). However, I would keep the isASCIILetter function in case we need it in the future, since it's already tested. wdyt?

Reply:
no need to bloat code with stuff we won't need. We will always have reference to this in github history, lets remove it for now.